lib/errors/aggregate-error.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const BaseError = require('./base-error');
+
+let level = 0;
+
+/**
+ * Validation Error. Thrown when the sequelize validation has failed. The error contains an `errors` property,
+ * which is an array with 1 or more ValidationErrorItems, one for each validation that failed.
+ *
+ * @param {string} message Error message
+ * @param {Array} [errors] Array of ValidationErrorItem objects describing the validation errors
+ *
+ * @property errors {ValidationErrorItems[]}
+ */
+class AggregateError extends BaseError {
+  constructor(message) {
+    super(message);
+    this.name = 'AggregateError';
+    Error.captureStackTrace(this, this.constructor);
+  }
+
+  toString() {
+    let indent = Array(level * 4 + 1).join(' ');
+    let ret = `\n${indent}AggregateError of:\n`;
+    level++;
+    indent = Array(level * 4 + 1).join(' ');
+    for (let i = 0; i < this.length; ++i) {
+      let str = this[i] === this ? '[Circular AggregateError]' : `${this[i] }`;
+      const lines = str.split('\n');
+      for (let j = 0; j < lines.length; ++j) {
+        lines[j] = indent + lines[j];
+      }
+      str = lines.join('\n');
+      ret += `${str }\n`;
+    }
+    level--;
+    return ret;
+  }
+}
+
+module.exports = AggregateError;
```

Comment:
Can we make all internal exports use `exports.NAME = NAME`? This is easier to model in es modules.

Reply:
@SimonSchick I got rid of this anyway, but it's not really hard with `jscodeshift` to convert all `module.exports = ` statements to `export default` statements.

Reply:
Oh wait, wrong PR, I haven't gotten rid of this error

Reply:
@jedwards1211 Please avoid default exports, see https://humanwhocodes.com/blog/2019/01/stop-using-default-exports-javascript-module/ as an example why.

Reply:
sigh, that would only complicate the conversion of everything to `export default`

Reply:
Or are y'all against `import default` and `export default` as well?

Reply:
export/import should generally be avoided, it leads to incosistent naming across modules, see https://google.github.io/styleguide/jsguide.html#named-vs-default-exports

Reply:
# 🙄

Reply:
I hope you're at least not going to make users convert
```js
import Sequelize from 'sequelize'
// or
const Sequelize = require('sequelize')
```
to
```js
import { Sequelize } from 'sequelize'
// 
const { Sequelize } = require('sequelize')
```

Reply:
And again, it we keep the code more consistent now, it's easier to automatically convert to whatever new form we want later

Reply:
> The only nitpicky rules I respect are ones that can be enforced through scripts with an automated fix, because that doesn't waste developers' time. 

Point taken, I will around if there are eslint rules for this in the future.

Reply:
apparently this does exist already, can't tell if it has an autofix or not?
https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-named-as-default.md

Reply:
Let's deal with this in another PR, there are opportunities to streamline exports/imports across the codebase.

Reply:
Yeah, I don't mind helping convert them automatically to either default or named.  (with the exception of the main export, I think we should leave that as is to minimize disruption for users)

Reply:
Man I feel bad for being such a grouch about this, I see you weren't having a great day with trying to debug the test failures, I imagine I made it worse 😕. Let me know what other stuff I can help with.

Reply:
Nah, you have a fair point with me being nit-picky about things that aren't in the lint configs trying to micro-manage these things in PRs isn't particularly productive and can be frustrating.